File: backend/ngts/src/main/java/com/three/ngts/Service/BusReviewService.java
package com.three.ngts.Service;

import com.three.ngts.Entity.Bus;
import com.three.ngts.Entity.BusReview;
import com.three.ngts.Entity.CustomEntity.TicketView;
import com.three.ngts.Repo.BusReviewRepo;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@CrossOrigin(origins = "http://localhost:3000")
@RestController
@RequestMapping("/api/")
public class BusReviewService {

    @Autowired
    private BusReviewRepo busReviewRepo;

    @PostMapping("/busReviews/insert")
    public BusReview insert(@RequestBody BusReview busReview) {
        return busReviewRepo.save(busReview);
    }

    @PostMapping("/busReviews/checkReview")
    public boolean checkReview(@RequestBody TicketView ticketView) {
        List<BusReview> busReviews = busReviewRepo.findAllByTicketAndReviewerAndBus(ticketView.getTicket(),
                ticketView.getTicket().getUser(), ticketView.getBusSchedule().getBus());
        return !busReviews.isEmpty();
    }

    @PostMapping("/busReviews/getBusReview")
    public List<BusReview> getBusReview(@RequestBody Bus bus) {
        return busReviewRepo.findAllByBus(bus);
    }
}

File: frontend/next-generation-ticketing-system/src/Service/BusReviewService.ts
import axios from "axios";

const USER_API_BASE_URL = "http://localhost:8080/api/busReviews";

class BusReviewService {
  insert(busReview: any) {
    return axios.post(USER_API_BASE_URL + "/insert", busReview);
  }

  checkReview(ticketView: any) {
    return axios.post(USER_API_BASE_URL + "/checkReview", ticketView);
  }

  getBusReview(bus: any) {
    return axios.post(USER_API_BASE_URL + "/getBusReview", bus);
  }
}

export default new BusReviewService();
